### PR TITLE
Misc docs and tests for runtime

### PR DESCRIPTION
**Pull request by teozkr**

### CODE REVIEW

File: kube-runtime/src/scheduler.rs [hunk 1]
```diff
@@ -126,9 +138,112 @@ where
...
+                message: 2,
+                run_at: Instant::now() + Duration::from_secs(3),
+            },
+        ]));
+        assert!(poll!(scheduler.next()).is_pending());
+        advance(Duration::from_secs(2)).await;
+        assert_eq!(scheduler.next().now_or_never().unwrap().unwrap().unwrap(), 1);
+        assert!(poll!(scheduler.next()).is_pending());
+        advance(Duration::from_secs(2)).await;
+        assert_eq!(scheduler.next().now_or_never().unwrap().unwrap().unwrap(), 2);
```
Comment (clux) [inline kube-runtime/src/scheduler.rs:176]: me as a kid at xmas

File: kube-runtime/src/scheduler.rs [hunk 2]
```diff
@@ -126,9 +138,112 @@ where
...
+                message: (),
+                run_at: Instant::now() + Duration::from_secs(3),
+            },
+        ]));
+        assert!(poll!(scheduler.next()).is_pending());
+        advance(Duration::from_secs(2)).await;
+        assert_eq!(scheduler.next().now_or_never().unwrap().unwrap().unwrap(), ());
+        // Stream has terminated
+        assert!(scheduler.next().await.is_none());
+    }
```
Comment (clux) [inline kube-runtime/src/scheduler.rs:199]: These tests are great.

File: kube-runtime/src/scheduler.rs [hunk 3]
```diff
@@ -126,9 +138,112 @@ where
...
+                run_at: Instant::now() + Duration::from_secs(1),
+            },
+            ScheduleRequest {
+                message: 2,
+                run_at: Instant::now() + Duration::from_secs(3),
+            },
+        ]));
+        assert!(poll!(scheduler.next()).is_pending());
+        advance(Duration::from_secs(2)).await;
+        assert_eq!(scheduler.next().now_or_never().unwrap().unwrap().unwrap(), 1);
```
Comment (clux) [inline kube-runtime/src/scheduler.rs:173]: is this a future of result of something? That's a lot of unwrapping

Comment (teozkr) [reply kube-runtime/src/scheduler.rs:173]: 1. The internal timer is fallible, so we wrap the `message` in a `Result<T>`
2. `Stream`s can be finite, so `StreamExt::next()` wraps the item as an `Option<Stream::Item>` (returning `None` once the `Stream` is exhausted)
3. `FutureExt::now_or_never()` needs to be able to signal that the future isn't completed yet, so it wraps it as an `Option<Future::Output>` (returning `None` if the `Future` hasn't completed yet)

Normally application code would use `Stream` composition instead, in which case they'd only deal with 1 (the `Result`)

**Pull request merged**